### PR TITLE
test: replace mocks with snapshots

### PR DESCRIPTION
**Pull request by markov00**
## Summary

We use mocks for checking the equality of very big objects.
The PR refactored that code using snapshot instead: they provide the same functionality with less burden.

## Details

This can simplify the refactoring in the following cases:
- a field is removed or slightly changed
- a completely new complex field is added. In particular, it is helpful when the new field is derived from others part of the data structure
- The mocks are themself code with some logic. They are not a guarantee to be bug-free.

If we are testing an object structure, we probably can benefit directly from snapshots instead of manually writing the object.
The only caveat is to threat snapshots as code, checking them during the review as they where written inline in the test

### CODE REVIEW

**Review by rshen91 — APPROVED** [review]
LGTM - This changing to snapshots makes sense to me especially in terms of the context of changing the data logic for the null values in tooltips PR. I think it should be manageable as a team for us to check any snapshot changes in a PR in future reviews (to make sure snapshots aren't being changed for bad reasons).

File: packages/charts/src/chart_types/xy_chart/rendering/rendering.bands.test.ts [hunk 1]
```diff
@@ -70,84 +68,11 @@ describe('Rendering bands - areas', () => {
...
-            x: 25,
-            y: 0,
-          },
-        }),
-      );
+      expect(points).toHaveLength(4);
+      expect(points[0]).toMatchSnapshot();
+      expect(points[1]).toMatchSnapshot();
+      expect(points[2]).toMatchSnapshot();
+      expect(points[3]).toMatchSnapshot();
```
Comment (nickofthyme) [inline packages/charts/src/chart_types/xy_chart/rendering/rendering.bands.test.ts:75]: Can't we simplify this? I don't see a reason for individual snapshots.

```suggestion
      expect(points).toMatchSnapshot();
```

**Review by nickofthyme — APPROVED** [review]
LGTM

File: packages/charts/src/chart_types/xy_chart/rendering/rendering.bands.test.ts (hunk 1 above)
Comment (markov00) [reply packages/charts/src/chart_types/xy_chart/rendering/rendering.bands.test.ts]: done in https://github.com/elastic/elastic-charts/pull/1220/commits/46aa1c830ba048346a9e4a4fb1e36d38512ef59f

**Pull request merged**